pkg/storageincentives/nodestatus.go

```diff
@@ -0,0 +1,121 @@
+// Copyright 2023 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package storageincentives
+
+import (
+	"context"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethersphere/bee/pkg/log"
+	"github.com/ethersphere/bee/pkg/settlement/swap/erc20"
+	"github.com/ethersphere/bee/pkg/storage"
+	"github.com/ethersphere/bee/pkg/swarm"
+	"math/big"
+	"sync"
+)
+
+const (
+	redistributionStatusKey = "redistribution_state_"
+)
+
+type NodeState struct {
+	stateStore     storage.StateStorer
+	erc20Service   erc20.Service
+	logger         log.Logger
+	overlay        swarm.Address
+	mtx            sync.Mutex
+	nodeStatus     NodeStatus
+	initialBalance *big.Int
+}
+
+// NodeStatus provide internal status of the nodes in the redistribution game
+type NodeStatus struct {
+	State  PhaseType `json:"state"`
+	Round  uint64    `json:"round"`
+	Block  uint64    `json:"block"`
+	Reward *big.Int  `json:"reward"`
+	Fees   *big.Int  `json:"fees"`
+}
+
+func NewNode(logger log.Logger, stateStore storage.StateStorer, erc20Service erc20.Service) NodeState {
+	return NodeState{
+		stateStore:     stateStore,
+		erc20Service:   erc20Service,
+		logger:         logger.WithName("nodestatus").Register(),
```

Comment:
Follow the convention it the other .go files and declare the logger name as constant.